pkcs11/yubihsm_pkcs11.c

```diff
@@ -1125,28 +1145,84 @@ CK_DEFINE_FUNCTION(CK_RV, C_Login)
   }
 
   yh_rc yrc;
-  yrc =
-    yh_create_session_derived(session->slot->connector, key_id, pPin, ulPinLen,
-                              true, &session->slot->device_session);
-  if (yrc != YHR_SUCCESS) {
-    DBG_ERR("Failed to create session: %s", yh_strerror(yrc));
-    if (yrc == YHR_CRYPTOGRAM_MISMATCH) {
-      rv = CKR_PIN_INCORRECT;
-    } else {
+
+  if (prefix == '@') { // Asymmetric authentication
+
+    uint8_t sk_oce[32], pk_oce[65], pk_sd[65];
+    size_t pk_sd_len = sizeof(pk_sd);
+    yrc = yh_util_derive_ec_p256_key(pPin, ulPinLen, sk_oce, sizeof(sk_oce),
+                                     pk_oce, sizeof(pk_oce));
+    if (yrc != YHR_SUCCESS) {
+      DBG_ERR("Failed to derive asymmetrix key: %s", yh_strerror(yrc));
```

Comment:
typo.

Reply:
Fixed in 3ce940dca5a0a563b3bc836119e7eba43b6182c4